src/__tests__/http2.test.ts

```diff
@@ -0,0 +1,83 @@
+import Koa from 'koa';
+import http2 from 'http2';
+import { ApolloServer } from '@apollo/server';
+import { ApolloServerPluginDrainHttpServer } from '@apollo/server/plugin/drainHttpServer';
+import cors from '@koa/cors';
+import bodyParser from 'koa-bodyparser';
+import { koaMiddleware } from '..';
+import { urlForHttpServer } from '../utils';
+
+describe('http2', () => {
+  it('works', async () => {
+    const app = new Koa();
+    // disable logs to console.error
+    app.silent = true;
+
+    // For browser-based implementations, you need to provide a key and cert and
+    // use http2.createSecureServer. This is because browsers will only use
+    // http2 with SSL.
+    const http2Server = http2.createServer(app.callback());
+    const server = new ApolloServer({
+      typeDefs: `#graphql
+        type Query {
+          hello: String!
+        }  
+      `,
+      resolvers: {
+        Query: {
+          hello: () => 'hello world!',
+        },
+      },
+      plugins: [
+        ApolloServerPluginDrainHttpServer({
+          // @ts-expect-error drain server plugin currently only accepts an http.Server
```

Comment:
Is the API close enough that this works? Wonder if it might be worth just throwing together an http2 drain plugin instead. You'd still have to do "track active sessions" yourself but within a given session you don't have to track active requests because the http2 API provides a graceful `http2session.close()` for you.